internal/signer/darwin/keychain/keychain.go

```
@@ -405,3 +443,123 @@ func certIn(xc *x509.Certificate, xcs []*x509.Certificate) bool {
+	sizeLim := uint64(C.SecKeyGetBlockSize(k.publicKeyRef)) - uint64(k.getPaddingSize())
+	if uint64(len(plaintext)) >= sizeLim {
+		return fmt.Errorf("plaintext is too long")
+	}
+	return nil
+}
+
+func (k *Key) getRSAEncryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	var algorithms map[crypto.Hash]C.CFStringRef
+	switch pub := k.Public().(type) {
+	case *rsa.PublicKey:
+		if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256) == 1 {
+			algorithms = rsaPSSAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256) == 1 {
+			algorithms = rsaOAEPAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256) == 1 {
+			algorithms = rsaPKCS1v15Algorithms
+		}
+	default:
+		return UNKNOWN_SECKEY_ALGORITHM, fmt.Errorf("algorithm is unsupported. only RSA algorithms are supported. %T", pub)
+	}
+	return algorithms[k.hash], nil
+}
+
+func (k *Key) getEncryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	if k.hash == 0 {
+		k.hash = crypto.SHA256
+	}
+	return k.getRSAEncryptAlgorithm()
+}
+
+func (k *Key) getRSADecryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	var algorithms map[crypto.Hash]C.CFStringRef
+	switch pub := k.Public().(type) {
+	case *rsa.PublicKey:
+		if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256) == 1 {
+			algorithms = rsaPSSAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256) == 1 {
+			algorithms = rsaOAEPAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256) == 1 {
+			algorithms = rsaPKCS1v15Algorithms
+		}
```

Similar comment. Else statement should return an error